test_pull_requests

```diff
@@ -359,6 +361,78 @@ def compose_bot_comment(prefix, comment_data={})
   updated_comment
 end
 
+class FlushableStore
+  def initialize(c_file=nil)
+    @cache_file = nil
+    @cache = {}
+    if c_file
+      @cache_file = File.expand_path(c_file)
+      if File.exists? @cache_file
```

Comment:
Do we really need to be manually checking if it exists/is writeable? Won't we see those errors from accessing it? There's some benefit from the "nicer" error messages here but I think I would prefer a method like:

```ruby
def initialize(cache_location)
    @cache_file = Marshal.load IO.read(File.expand_path(cache_location)))
end
```